### PR TITLE
Added .gitignore in archtype

### PR DESCRIPTION
**Pull request by vapadwal**
Please see https://github.com/devonfw/devon4j/issues/258

### CODE REVIEW

File: templates/server/pom.xml [hunk 1]
```diff
@@ -37,6 +37,8 @@
           <delimiters>
             <delimiter>$[*]</delimiter>
           </delimiters>
+          <!--Requires below statement, please see https://issues.apache.org/jira/browse/MRESOURCES-190 -->
+          <addDefaultExcludes>false</addDefaultExcludes>
```
Comment (hohwille) [inline templates/server/pom.xml:41]: We need to test this. The explicit `include` is not enough? Disabling the default excludes may include some other undesired files into the archetype. Maybe we can create archetype before and after this change and diff the files included.

Comment (vapadwal) [reply templates/server/pom.xml:41]: yes have ask @Rohit90700  also to test this at his end

Comment (Rohit90700) [reply templates/server/pom.xml:41]: Hi @vapadwal  , I have tested ,its working fine for me.

Comment (hohwille) [reply templates/server/pom.xml:41]: @Rohit90700 thanks for testing so far.
I am not in doubt that with the change `.gitignore` will be added but will also other undesired files (`*.bak`, `*.log`,`.project`, ...) be added accidentally in the archetype?
Did you diff the outcome before and after this change as suggested in my comment above?

Comment (Rohit90700) [reply templates/server/pom.xml:41]: Hi @hohwille , I have tested both scenario before and after the change as you suggested above comment. there is only one file difference of .gitignore .

**Comment by hohwille** [comment]
Ready to be merged as soon as updated and green.

**Pull request merged**